packages/editor/src/store/selectors.js

```diff
@@ -388,8 +388,9 @@ export function isEditedPostEmpty( state ) {
  * @return {boolean} Whether the post can be autosaved.
  */
 export function isEditedPostAutosaveable( state ) {
-	// A post must contain a title, an excerpt, or non-empty content to be valid for autosaving.
-	if ( ! isEditedPostSaveable( state ) ) {
+	// A post needs to be in dirty state and,
+	// a post must contain a title, an excerpt, or non-empty content to be valid for autosaving.
+	if ( ! isEditedPostDirty( state ) || ! isEditedPostSaveable( state ) ) {
```

Comment:
I noticed this comment previously about why `isEditedPostDirty` can't be used in `isEditedPostSaveable` and wondered whether it applies here as well:
https://github.com/WordPress/gutenberg/blob/1349d232b25ea6573931589023d7b42bffd85eb9/packages/editor/src/store/selectors.js#L346